Kernel/Net/NetworkAdapter.h

```diff
@@ -45,6 +45,8 @@ class NetworkAdapter : public Weakable<NetworkAdapter> {
     u32 packets_out() const { return m_packets_out; }
     u32 bytes_out() const { return m_bytes_out; }
 
+    void set_on_receive(Function<void()> on_receive) { m_on_receive = move(on_receive); }
```

Comment:
The convention throughout the codebase is to just have a public member:
```cpp
Function<void()> on_receive;
```

Reply:
Easy change - will do that now.